src/plugins/condition/components/Criterion.vue

```diff
@@ -276,17 +279,17 @@ export default {
                 if (!this.filteredOps.find(operation => operation.name === this.criterion.operation)) {
                     this.criterion.operation = '';
                     this.criterion.input = this.enumerations.length ? [this.enumerations[0].value.toString()] : [];
-                    this.inputCount = 0;
                 }
             } else {
                 if (this.enumerations.length && !this.criterion.input.length) {
                     this.criterion.input = [this.enumerations[0].value.toString()];
                 }
-                this.inputCount = 0;
             }
         },
         persist() {
-            this.$emit('persist', this.criterion);
+            if (this.criterion.telemetry && this.criterion.metadata !== '' && this.criterion.operation !== '') {
+                this.$emit('persist', this.criterion);
+            }
```

Comment:
Question. Is this related or just an optimization you spotted while working on this? Also, I don't know the answer to this question, but just making sure if this is the correct check before persisting? I would think if you wanted to not persist, the check would be to see if the state changed, and only persist if it has. This seems to be checking if there is a certain state and only persist in that case. Plus, I don't think criterion.metadata and criterion.operation are ever `''` are they? I think they could only be something, or`undefined`. Maybe `null` if we set it up that way.

Reply:
It was an attempt to solve the problem that was not successful, so removed it.